### PR TITLE
Navigation 🔥

### PR DESCRIPTION
**Pull request by moralesbang**
### Summary

Added navigation to book detail screen using `react-navigation`.

### Screenshots

react-navigation-1

### Trello Card

https://trello.com/c/QVTWSrtN

### CODE REVIEW

File: src/app/components/BookCard/index.tsx [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 import styles from './styles';
 
 interface Props extends Book {}
 
-function BookCard({ title, author, imageUrl }: Props) {
+function BookCard({ id, title, author, imageUrl }: Props) {
+  const navigation: StackNavigationProp<RootStackParamList, Routes.Library> = useNavigation();
   return (
-    <View style={styles.book}>
+    <TouchableOpacity style={styles.book} onPress={() => navigation.push(Routes.BookDetail, { bookId: id })}>
```
Comment (marianozicavo) [inline src/app/components/BookCard/index.tsx:17]: Let's extract this onPress function

Comment (marianozicavo) [inline src/app/components/BookCard/index.tsx:17]: Let's also use navigation.navigate here

Comment (moralesbang) [reply src/app/components/BookCard/index.tsx:17]: Ok, sounds good.

Comment (moralesbang) [reply src/app/components/BookCard/index.tsx:17]: 💍

File: src/app/components/BookCard/index.tsx [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
+import Routes from '@constants/routes';
 
 import styles from './styles';
 
 interface Props extends Book {}
 
-function BookCard({ title, author, imageUrl }: Props) {
+function BookCard({ id, title, author, imageUrl }: Props) {
+  const navigation: StackNavigationProp<RootStackParamList, Routes.Library> = useNavigation();
+  const handlePress = () => navigation.navigate(Routes.BookDetail, { bookId: id });
```
Comment (marianozicavo) [inline src/app/components/BookCard/index.tsx:16]: Pass the whole book here, so you don't have to find it in the BookDetail

File: src/app/screens/BookDetail/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import bookPlaceholderCover from '@assets/book_placeholder_cover.png';
+
+import styles from './styles';
+
+interface Props {
+  route: RouteProp<RootStackParamList, Routes.BookDetail>;
+}
+
+function BookDetail({ route }: Props) {
+  const selectedBook = BOOKS.find(({ id }) => id === route.params.bookId);
```
Comment (marianozicavo) [inline src/app/screens/BookDetail/index.tsx:16]: Use navigation.getParams to get the navigation parameters

Comment (marianozicavo) [reply src/app/screens/BookDetail/index.tsx:16]: Actually, over the weekend I learned this is not the case any more with react navigation 5. Dismiss this comment 😅

**Pull request merged**